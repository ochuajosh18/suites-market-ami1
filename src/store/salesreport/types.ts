export type DynamicBasicReportType = 
string | 
boolean |
Array<ReportValuePair> |
Array<unknown> |
CallVisitData |
TopSellingProductsData |
Array<SalespersonCallVisitData> |
TotalSalesData |
undefined;
export type ReportValuePair = { label: string, value: string };
export type ReportTab = 'SUMMARY' | 'TRANSACTION';

// summary report types
export interface CallVisitData {
    complete: number;
    completePercentage: number;
    incomplete: number;
    incompletePercentage: number;
    total: number;
}

export interface TopSellingProductsData {
    products: Array<{ name: string; qty: number; percentage: number }>;
    total: number;
}

export interface TotalSalesData {
    totalSales: number;
    difference: number;
    differencePercentage: number;
    monthlySales: Array<{ month: string; total: number; }>;
}

export interface SalespersonCallVisitData {
    count: number;
    salespersonName: string;
}

export interface BasicReportType<T> {
    [key: string]: T;
}

export interface BasicReportStateInput {
    [name: string]: DynamicBasicReportType
}

export interface ReportState extends BasicReportType<DynamicBasicReportType> {
    activeTab: ReportTab;
    reportDateStart: string;
    reportDateEnd: string;
    reportType: string;
    reportSalesperson: string;
    reportSalespersonList: Array<ReportValuePair>;
    reportData: Array<unknown>
    reportLoading: boolean;
    callVisitSummaryLoading: boolean;
    callVisitData?: CallVisitData;
    callVisitMonth: string;
    topSellingProductsLoading: boolean;
    topSellingProductsData?: TopSellingProductsData;
    topSellingProductsMonth: string;
    salespersonCallVisitLoading: boolean;
    salespersonCallVisitData: Array<SalespersonCallVisitData>;
    salespersonCallVisitMonth: string;
    totalSalesLoading: boolean;
    totalSalesData?: TotalSalesData;
    totalSalesMonthFrom: string;
    totalSalesMonthTo: string;
    totalSalesYear: string;
}

export const SET_REPORT_STATE = 'set_report_state';

export interface setReportStateAction {
    type: typeof SET_REPORT_STATE;
    payload: BasicReportStateInput;
}

export type ReportAction = setReportStateAction;
