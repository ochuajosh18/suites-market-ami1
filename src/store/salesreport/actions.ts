import {
    SET_REPORT_STATE,
    BasicReportStateInput,
    ReportAction
} from './types';
import { AppThunk } from '..';
import axios from 'axios';
import map from 'lodash/map';
import moment from 'moment';
const API_URL = process.env.REACT_APP_API_URL;

export const setReportState = (input: BasicReportStateInput): ReportAction => ({
    type: SET_REPORT_STATE,
    payload: input
});


export const loadSalespersonList = (): AppThunk =>  {
    return async (dispatch) => {
        try {
            const res = await axios.get( `${API_URL}/user/salesperson`);
            if (res.data) {
                dispatch({
                    type: SET_REPORT_STATE,
                    payload: { 
                        reportSalespersonList: map(res.data, (salesperson) => ({
                            label: `${salesperson.displayId} - ${salesperson.firstName} ${salesperson.lastName}`,
                            value: salesperson.id
                        }))
                    }
                })
            }
        }
        catch (e) {
            console.log(e);
        }
        finally {

        }
    }
}

export const generateReport = (): AppThunk => {
    return async (dispatch, getState) => {
        const { activeTab, reportType,  reportSalesperson, reportDateStart, reportDateEnd } = getState().salesreport;
        dispatch({
            type: SET_REPORT_STATE,
            payload: { reportLoading: true, reportData: [] }
        });
        const TYPES = {
            CALL_VISIT_MEETING: 'meeting',
            CALL_VISIT_PROFILING: 'customer-profiling',
            TFO: 'tfo',
            CUSTOMER: 'customer'
        }
        try {
            const queryParamObj = {
                salespersonId: reportSalesperson !== 'All' ? reportSalesperson : '',
                startDate: reportDateStart ? moment(reportDateStart, 'DD/MM/YYYY').format("YYYY-MM-DD") : '',
                endDate: reportDateEnd ? moment(reportDateEnd, 'DD/MM/YYYY').format("YYYY-MM-DD") : ''
            }
            let queryParams = '?';
            for (const key of Object.keys(queryParamObj)) {
                queryParams = queryParamObj[key] ? queryParams + `${key}=${queryParamObj[key]}&` : queryParams
            }
            const res = await axios.get( `${API_URL}/report/${activeTab.toLowerCase()}/${TYPES[reportType]}${queryParams}`);
            if (res.data) {
                dispatch({
                    type: SET_REPORT_STATE,
                    payload: { 
                        reportData: res.data
                    }
                })
            }
        }
        catch (e) {
            console.log(e);
        }
        finally {
            dispatch({
                type: SET_REPORT_STATE,
                payload: { reportLoading: false }
            });
        }
    }
}

export const getTotalCallsSummary = (month = moment().format('MM')): AppThunk => async (dispatch) => {
    dispatch({
        type: SET_REPORT_STATE,
        payload: { callVisitSummaryLoading: true }
    });
    
    try {
        const repRes = await axios.get(`${API_URL}/v1/reports/summary/call-visit?month=${month}`);
        if (repRes.status === 200) {
            dispatch({
                type: SET_REPORT_STATE,
                payload: { callVisitData: repRes.data }
            });
        }
    }
    catch (e) {

    }
    finally {
        dispatch({
            type: SET_REPORT_STATE,
            payload: { callVisitSummaryLoading: false }
        });
    }
}

export const getTopSellingProducts = (month = moment().format('MM')): AppThunk => async (dispatch) => {
    dispatch({
        type: SET_REPORT_STATE,
        payload: { topSellingProductsLoading: true }
    });
    
    try {
        const repRes = await axios.get(`${API_URL}/v1/reports/summary/top-selling-product?month=${month}`);
        if (repRes.status === 200) {
            dispatch({
                type: SET_REPORT_STATE,
                payload: { topSellingProductsData: repRes.data }
            });
        }
    }
    catch (e) {

    }
    finally {
        dispatch({
            type: SET_REPORT_STATE,
            payload: { topSellingProductsLoading: false }
        });
    }
}


export const getSalespersonCallVisitSummary = (month = moment().format('MM')): AppThunk => async (dispatch) => {
    dispatch({
        type: SET_REPORT_STATE,
        payload: { salespersonCallVisitLoading: true }
    });
    
    try {
        const repRes = await axios.get(`${API_URL}/v1/reports/summary/top-salesperson-call-visit?month=${month}`);
        if (repRes.status === 200) {
            dispatch({
                type: SET_REPORT_STATE,
                payload: { salespersonCallVisitData: repRes.data }
            });
        }
    }
    catch (e) {

    }
    finally {
        dispatch({
            type: SET_REPORT_STATE,
            payload: { salespersonCallVisitLoading: false }
        });
    }
}


export const getTotalSales = (year = moment().format('YYYY'), startMonth = '01', endMonth = '12'): AppThunk => async (dispatch) => {
    dispatch({
        type: SET_REPORT_STATE,
        payload: { totalSalesLoading: true }
    });
    
    try {
        const repRes = await axios.get(`${API_URL}/v1/reports/summary/total-sales?year=${year}&startMonth=${startMonth}&endMonth=${endMonth}`);
        if (repRes.status === 200) {
            dispatch({
                type: SET_REPORT_STATE,
                payload: { totalSalesData: repRes.data }
            });
        }
    }
    catch (e) {

    }
    finally {
        dispatch({
            type: SET_REPORT_STATE,
            payload: { totalSalesLoading: false }
        });
    }
}

export const initialSummaryReports = (): AppThunk => async (dispatch) => {
    dispatch(getTotalSales());
    dispatch(getTopSellingProducts());
    dispatch(getSalespersonCallVisitSummary());
    dispatch(getTotalCallsSummary());
}