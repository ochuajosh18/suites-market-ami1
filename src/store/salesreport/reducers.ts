import {
    SET_REPORT_STATE,
    ReportAction,
    ReportState
} from './types';

const INITIAL_STATE: ReportState = {
    activeTab: 'SUMMARY',
    reportDateStart: '',
    reportDateEnd: '',
    reportSalesperson: '',
    reportType: '',
    reportSalespersonList: [],
    reportData: [],
    reportLoading: false,
    callVisitSummaryLoading: false,
    callVisitMonth: '',
    topSellingProductsLoading: false,
    topSellingProductsMonth: '',
    salespersonCallVisitLoading: false,
    salespersonCallVisitMonth: '',
    salespersonCallVisitData: [],
    totalSalesLoading: false,
    totalSalesMonthFrom: '',
    totalSalesMonthTo: '',
    totalSalesYear: ''
}

export default (state = INITIAL_STATE, action: ReportAction): ReportState => {
    switch (action.type) {
        case SET_REPORT_STATE:
            return { ...state, ...action.payload };
        default:
            return state;
    }
}