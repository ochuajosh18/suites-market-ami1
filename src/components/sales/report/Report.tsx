import React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../../../store';
import { ReportState, ReportTab } from '../../../store/salesreport/types';
import { setReportState, loadSalespersonList, generateReport, initialSummaryReports } from '../../../store/salesreport/actions';
import { setSystemState } from '../../../store/system/actions';

import SummaryReport from './fragments/SummaryReport';
import ReportRenderer from './fragments/ReportRenderer';
import { ReportBaseContainer, ReportHeaderButton } from './fragments/ReportComponents';

import { 
    SymphonyTabs,
    SymphonyTab,
    SymphonyTabsContainer
} from '../../symphony/SymphonyCommonComponents';
import SymphonyLayout from '../../symphony/SymphonyLayout';
import { SYMPHONY_PRIMARY_COLOR } from '../../symphony/Colors';

import Icon from '@material-ui/core/Icon';
import { toastWarning } from '../../../modules/Toast';
import { CSVLink } from 'react-csv';

import Box from '@material-ui/core/Box';

interface ReportProps {
    setReportState: typeof setReportState;
    setSystemState: typeof setSystemState;
    loadSalespersonList: typeof loadSalespersonList;
    generateReport: typeof generateReport;
    initialSummaryReports: typeof initialSummaryReports;
    salesreport: ReportState;
}

class Report extends React.Component<ReportProps> {
    csvRef;

    constructor(props) {
        super(props);
        this.csvRef = React.createRef<typeof CSVLink>();
    }
    
    componentDidMount = () => {
        this.props.loadSalespersonList()
        // set button
        this.props.setSystemState({
            headerEndButton: () => (
                <Box display="flex">
                    <ReportHeaderButton 
                        endIcon={<Icon className="fas fa-undo" style={{ transform: 'rotate(-45deg)' }} />}
                        onClick={this._onResetClick.bind(this)}
                    >
                        Reset
                    </ReportHeaderButton>
                    <ReportHeaderButton 
                        endIcon={<Icon className="fas fa-file-export" />}
                        onClick={this._onExportClick.bind(this)}
                    >
                        Export
                    </ReportHeaderButton>
                </Box>
            )
        })
    }

    componentWillUnmount = () => {
        this.props.setSystemState({ headerEndButton: undefined })
    }

    _onTabClick = (value: ReportTab) => {
        this.props.setReportState({ activeTab: value, reportType: '', reportSalesperson: '', reportDateStart: '', reportDateEnd: '' });
    }

    _onReportInput = (field: string, value: string) => {
        this.props.setReportState({ [field]: value, reportData: [] })
    }

    _onViewClick = () => {
        const { reportType, reportSalesperson, reportDateStart } = this.props.salesreport;
        if (!reportType) { toastWarning("Please select a report type"); return; }
        if (!reportSalesperson) { toastWarning("Please select a salesperson"); return; }
        if (!reportDateStart) { toastWarning("Please indicate the start date"); return; }
        this.props.generateReport()
    }

    _onResetClick = () => {
        this.props.setReportState({ 
            reportData: [],
            reportType: '', 
            reportSalesperson: '', 
            reportDateStart: '', 
            reportDateEnd: '',
            totalSalesMonthFrom: '',
            totalSalesMonthTo: '',
            totalSalesYear: '',
            salespersonCallVisitMonth: '',
            topSellingProductsMonth: '',
            callVisitMonth: '',
            callVisitData: undefined,
            totalSalesData: undefined,
            salespersonCallVisitData: undefined,
            topSellingProductsData: undefined
        });
        this.props.initialSummaryReports();
    }

    _onExportClick = () => {
        const { reportData } = this.props.salesreport;
        if (reportData.length > 0) {
            if (this.csvRef.current) {
                this.csvRef.current.link.click()
            }
        }
        else {
            toastWarning("Please generate data before exporting ")
        }
    }

    render = () => {
        const { activeTab, reportDateStart, reportDateEnd, reportSalesperson, reportType, reportSalespersonList, reportData, reportLoading } = this.props.salesreport;
        return (
            <SymphonyLayout>
                <ReportBaseContainer style={{ backgroundColor: 'transparent'}}>
                    <SymphonyTabsContainer>
                        <SymphonyTabs
                            value={activeTab}
                            // style={{ padding: '20px 12px 0', backgroundColor: '#F4F6F9' }}
                            TabIndicatorProps={{ style: { height: 3, backgroundColor: SYMPHONY_PRIMARY_COLOR }}}
                        >
                            <SymphonyTab
                                label="Summary Report"
                                value="SUMMARY"
                                onClick={this._onTabClick.bind(this, 'SUMMARY')}
                            />
                            <SymphonyTab
                                label="Transaction Report"
                                value="TRANSACTION"
                                onClick={this._onTabClick.bind(this, 'TRANSACTION')}
                            /> 
                        </SymphonyTabs>
                    </SymphonyTabsContainer>
                    {activeTab === 'TRANSACTION' &&
                        <ReportRenderer  
                            dateStart={reportDateStart}
                            dateEnd={reportDateEnd}
                            salesperson={reportSalesperson}
                            type={reportType}
                            salespersonList={reportSalespersonList}
                            onReportInput={this._onReportInput.bind(this)}
                            activeTab={activeTab}
                            data={reportData}
                            onViewClick={this._onViewClick.bind(this)}
                            loading={reportLoading}
                            csvRef={this.csvRef}
                        />
                    }
                    {activeTab === 'SUMMARY' &&
                        <SummaryReport />
                    }
                </ReportBaseContainer>
            </SymphonyLayout>
        )
    }
}

const mapStateToProps = (state: AppState) => ({
    salesreport: state.salesreport
});

export default connect(mapStateToProps, {
    setReportState,
    setSystemState,
    loadSalespersonList,
    generateReport,
    initialSummaryReports
})(Report)