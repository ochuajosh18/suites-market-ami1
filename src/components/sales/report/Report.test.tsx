import React from 'react';
import { mount } from 'enzyme';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import salesreport from '../../../store/salesreport/reducers';
import Report from './Report';

const createTestStore = () => {
    return createStore(
        combineReducers({
            salesreport
        }),
        applyMiddleware(thunk)
    );
}

const setup = (store: ReturnType<typeof createTestStore>) => {
    let wrapper = mount(
        <Provider store={store}>
            <Report />
        </Provider>
    )
    return wrapper;
}


describe('Sales Report', () => {
    let store: ReturnType<typeof createTestStore>;
    
    it('renders the router and the report snapshot', () => {
        store = createTestStore();
        const wrapper = setup(store);
        expect(wrapper.render().html()).toMatchSnapshot()
    });

    it('renders the canvas for the reports', () => {
        store = createTestStore();
        const wrapper = setup(store);

        expect(wrapper.find('#call-visit-summary-report-container')).not.toHaveLength(0);
        expect(wrapper.find('#top-selling-products-report-container')).not.toHaveLength(0);
        expect(wrapper.find('#total-sales-report-container')).not.toHaveLength(0);
        expect(wrapper.find('#call-visit-per-salesperson-report-container')).not.toHaveLength(0);
    });
});