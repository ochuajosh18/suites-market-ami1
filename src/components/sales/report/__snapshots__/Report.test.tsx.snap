// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sales Report renders the router and the report snapshot 1`] = `"<div class=\\"MuiBox-root MuiBox-root-3 Styled(MuiBox)-root-2\\" classes=\\"[object Object]\\" style=\\"background-color: transparent;\\"><div class=\\"MuiBox-root MuiBox-root-5 Styled(MuiBox)-root-4\\" classes=\\"[object Object]\\"><div class=\\"MuiTabs-root WithStyles(ForwardRef(Tabs))-root-6\\"><div class=\\"MuiTabs-scroller MuiTabs-fixed\\" style=\\"overflow: hidden;\\"><div class=\\"MuiTabs-flexContainer\\" role=\\"tablist\\"><button class=\\"MuiButtonBase-root MuiTab-root WithStyles(ForwardRef(Tab))-root-7 MuiTab-textColorInherit Mui-selected\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"true\\"><span class=\\"MuiTab-wrapper\\">Summary Report</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiTab-root WithStyles(ForwardRef(Tab))-root-7 MuiTab-textColorInherit\\" tabindex=\\"-1\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\"><span class=\\"MuiTab-wrapper\\">Transaction Report</span><span class=\\"MuiTouchRipple-root\\"></span></button></div><span class=\\"PrivateTabIndicator-root-8 PrivateTabIndicator-colorSecondary-10 MuiTabs-indicator\\" style=\\"left: 0px; width: 0px; height: 3px; background-color: rgb(181, 121, 54);\\"></span></div></div></div><div class=\\"MuiBox-root MuiBox-root-13 Styled(MuiBox)-root-12\\" classes=\\"[object Object]\\"><div class=\\"MuiBox-root MuiBox-root-15 Styled(MuiBox)-root-14\\" classes=\\"[object Object]\\" id=\\"total-sales-report-container\\"><div class=\\"MuiBox-root MuiBox-root-16\\"><div class=\\"MuiBox-root MuiBox-root-17\\"><div class=\\"MuiBox-root MuiBox-root-18\\">Total Sales</div></div><div class=\\"MuiBox-root MuiBox-root-19\\"><div class=\\"MuiBox-root MuiBox-root-20\\" style=\\"position: relative; display: flex; justify-content: center; align-items: center; height: 12px;\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text report-aux-button\\" tabindex=\\"0\\" type=\\"button\\" style=\\"padding: 0px; color: rgb(181, 121, 54); margin-right: 16px; text-transform: none;\\"><span class=\\"MuiButton-label\\">2021<span class=\\"MuiButton-endIcon MuiButton-iconSizeMedium\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color=\\"#B57936\\" aria-hidden=\\"true\\"><path d=\\"M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z\\"></path></svg></span></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text report-aux-button\\" tabindex=\\"0\\" type=\\"button\\" style=\\"padding: 0px; color: rgb(181, 121, 54); margin-right: 16px; text-transform: none;\\"><span class=\\"MuiButton-label\\">January<span class=\\"MuiButton-endIcon MuiButton-iconSizeMedium\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color=\\"#B57936\\" aria-hidden=\\"true\\"><path d=\\"M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z\\"></path></svg></span></span><span class=\\"MuiTouchRipple-root\\"></span></button><div class=\\"MuiBox-root MuiBox-root-21\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color=\\"#B57936\\" aria-hidden=\\"true\\"><path d=\\"M19 13H5v-2h14v2z\\"></path></svg></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text report-aux-button\\" tabindex=\\"0\\" type=\\"button\\" style=\\"padding: 0px; color: rgb(181, 121, 54); text-transform: none;\\"><span class=\\"MuiButton-label\\">December<span class=\\"MuiButton-endIcon MuiButton-iconSizeMedium\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color=\\"#B57936\\" aria-hidden=\\"true\\"><path d=\\"M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z\\"></path></svg></span></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div class=\\"MuiBox-root MuiBox-root-37\\"><div class=\\"MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width: 40px; height: 40px; color: rgb(181, 121, 54);\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div></div><div class=\\"MuiBox-root MuiBox-root-22 Styled(MuiBox)-root-14\\" classes=\\"[object Object]\\" id=\\"call-visit-summary-report-container\\"><div class=\\"MuiBox-root MuiBox-root-23\\"><div class=\\"MuiBox-root MuiBox-root-24\\">Total Call Visits</div><div style=\\"position: relative; display: flex; justify-content: center; align-items: center; margin-right: 8px;\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text report-aux-button\\" tabindex=\\"0\\" type=\\"button\\" style=\\"padding: 0px; color: rgb(181, 121, 54); position: absolute; right: -10px; width: 24px; height: 24px; text-transform: none;\\"><span class=\\"MuiButton-label\\">July<span class=\\"MuiButton-endIcon MuiButton-iconSizeMedium\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color=\\"#B57936\\" aria-hidden=\\"true\\"><path d=\\"M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z\\"></path></svg></span></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div><div class=\\"MuiBox-root MuiBox-root-25\\"><div class=\\"MuiBox-root MuiBox-root-38\\"><div class=\\"MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width: 40px; height: 40px; color: rgb(181, 121, 54);\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div></div></div><div class=\\"MuiBox-root MuiBox-root-26 Styled(MuiBox)-root-14\\" classes=\\"[object Object]\\" id=\\"top-selling-products-report-container\\"><div class=\\"MuiBox-root MuiBox-root-27\\"><div class=\\"MuiBox-root MuiBox-root-28\\"><div class=\\"MuiBox-root MuiBox-root-29\\">Top Selling Products</div><div class=\\"MuiBox-root MuiBox-root-30\\">Based on QTY Sold</div></div><div style=\\"position: relative; display: flex; justify-content: center; align-items: center; height: 12px;\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text report-aux-button\\" tabindex=\\"0\\" type=\\"button\\" style=\\"padding: 0px; color: rgb(181, 121, 54); position: absolute; right: -10px; text-transform: none;\\"><span class=\\"MuiButton-label\\">July<span class=\\"MuiButton-endIcon MuiButton-iconSizeMedium\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color=\\"#B57936\\" aria-hidden=\\"true\\"><path d=\\"M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z\\"></path></svg></span></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div><div class=\\"MuiBox-root MuiBox-root-39\\"><div class=\\"MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width: 40px; height: 40px; color: rgb(181, 121, 54);\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div></div><div class=\\"MuiBox-root MuiBox-root-31 Styled(MuiBox)-root-14\\" classes=\\"[object Object]\\" id=\\"call-visit-per-salesperson-report-container\\"><div class=\\"MuiBox-root MuiBox-root-32\\"><div class=\\"MuiBox-root MuiBox-root-33\\">Top 5 - Call Visits Made per Salesperson</div><div style=\\"position: relative; display: flex; justify-content: center; align-items: center; margin-right: 8px;\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text report-aux-button\\" tabindex=\\"0\\" type=\\"button\\" style=\\"padding: 0px; color: rgb(181, 121, 54); position: absolute; right: -10px; width: 24px; height: 24px; text-transform: none;\\"><span class=\\"MuiButton-label\\">July<span class=\\"MuiButton-endIcon MuiButton-iconSizeMedium\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color=\\"#B57936\\" aria-hidden=\\"true\\"><path d=\\"M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z\\"></path></svg></span></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div><div class=\\"MuiBox-root MuiBox-root-34\\"><div class=\\"MuiBox-root MuiBox-root-40\\"><div class=\\"MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width: 40px; height: 40px; color: rgb(181, 121, 54);\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div></div></div></div></div><div class=\\"Toastify\\"></div>"`;
